skymp5-server/cpp/localization_provider/LocalizationProvider.h

```diff
@@ -0,0 +1,36 @@
+#include <filesystem>
+#include <map>
+#include <string>
+#include <vector>
+
+class DirectoryEntry
+{
+public:
+  uint32_t stringId;
+  uint32_t offset;
+  uint32_t length;
+  std::string str;
+
+  DirectoryEntry();
+  DirectoryEntry(uint32_t stringId, uint32_t stringOffset);
+};
+
+class LocalizationProvider
+{
+  std::vector<DirectoryEntry*> ParseDirectoryEntries(std::vector<char> buffer);
```

Comment:
Bad design. Raw pointers used such way are road to memory leaks.

```suggestion
  std::vector<shared_ptr<DirectoryEntry> > ParseDirectoryEntries(const std::vector<char> &buffer);
```

```suggestion
  void ParseDirectoryEntries(const std::vector<char> &buffer, std::vector<DirectoryEntry> &container);
```